src/rpc/connection.rs

```diff
@@ -568,8 +568,9 @@ impl CircularQueue {
         let mut start_index = self.start_index.lock().unwrap();
         let end_flag = self.end_index.load(Ordering::Relaxed);
         for i in *start_index..end_flag {
```

Comment:
Maybe it should use `while` rather than `for`.

Reply:
why is `while` better?

Reply:
why not use atomic operation for start_index?

Reply:
if the start_index was overwritten by the old value, it may cause some problems. 

> why not use atomic operation for start_index?

I think it is more important to keep the count of registered id under the capacity of queue. And then the operation can be optimized reasonably.

Reply:
I don't understand it. Plz give some examples.